pygmt/__init__.py

```diff
@@ -110,8 +113,18 @@ def show_versions(file=sys.stdout):
     import subprocess
 
     from packaging.requirements import Requirement
+    from packaging.version import Version
+
+    def _get_clib_info() -> dict:
```

Comment:
The `pygmt.print_clib_info` function was initially added in #176 but has been replaced by `pygmt.show_versions` in #466 (first appear in v0.1.2).

In this PR, a new private function `_get_clib_info` is added which returns a dictionary instead of printing a long string. 

It's time to retire the `print_clib_info` function. Since it's rarely used, I think we can just remove it without a deprecation warning. Of course, we should do it in a separate PR so that at least we have a deprecation entry in the v0.13.0 changelog.